### ISSUE TITLE
OrthoSlicer creating axes for unused views

### ISSUE THREAD

**Issue by jeromedockes** [nilearn/nilearn#2202]
the `OrthoSlicer` seems to create and allocate the same space to three axes,
even when one view is required.
this can result in lost space, see this small example.

```python
from nilearn import plotting, datasets
from matplotlib import pyplot as plt

mni = datasets.load_mni152_template()
fig, ax = plt.subplots(figsize=(5, 5))
plotting.plot_glass_brain(mni, axes=ax, display_mode="z")

fig.savefig("/tmp/glass_brain.pdf", bbox_inches="tight", facecolor="gray")
```

result: 
![glass_brain](https://user-images.githubusercontent.com/9196501/68327258-47bbb100-00cd-11ea-9a1a-b885156adc5b.png)

it seems that changing [this line](https://github.com/nilearn/nilearn/blob/1a821a2b8620cf56491708ca84d0bc8b1e70d9ee/nilearn/plotting/displays.py#L1096) as shown below would fix the issue, but I haven't looked at it in depth yet

```diff
diff --git a/nilearn/plotting/displays.py b/nilearn/plotting/displays.py
index 5f25e54..1f5c502 100644
--- a/nilearn/plotting/displays.py
+++ b/nilearn/plotting/displays.py
@@ -1091,10 +1091,11 @@ class OrthoSlicer(BaseSlicer):
          facecolor = 'k' if self._black_bg else 'w'
          # Create our axes:
          self.axes = dict()
+        width = 1 / len(self._cut_displayed)
          for index, direction in enumerate(self._cut_displayed):
              fh = self.frame_axes.get_figure()
-            ax = fh.add_axes([0.3 * index * (x1 - x0) + x0, y0,
-                              .3 * (x1 - x0), y1 - y0], aspect='equal')
+            ax = fh.add_axes([width * index * (x1 - x0) + x0, y0,
+                              width * (x1 - x0), y1 - y0], aspect='equal')
              if LooseVersion(matplotlib.__version__) >= LooseVersion("1.6"):
                  ax.set_facecolor(facecolor)
              else:
```

fixed:

![glass_brain_1](https://user-images.githubusercontent.com/9196501/68327272-5013ec00-00cd-11ea-8ddc-24bfb812f406.png)

ping @GaelVaroquaux ?

**Comment by KamalakerDadi** [comment]
Looks ok to me. I tried your small example on my box and I cannot reproduce the first plot your showing.

**Comment by bthirion** [comment]
Same thing here, I don't reproduce you figure @jeromedockes

**Comment by kchawla-pi** [comment]
Is everyone using the same version?

**Comment by bthirion** [comment]
0.6.0a

**Comment by jeromedockes** [comment]
I realized I can only reproduce this with matplotlib 3.0.0 (with any
backend), not 3.1.1 .

I will try to see why the axes are not being rescaled correctly,
the
[`_locator`](https://github.com/nilearn/nilearn/blob/1a821a2b8620cf56491708ca84d0bc8b1e70d9ee/nilearn/plotting/displays.py#L1123)
should be taking care of this

**Comment by NicolasGensollen** [comment]
Hi @jeromedockes 
I took a look at this and I was able to reproduce the bug with matplotlib 3.0.0.
Here is the behavior that I observed:

- 3.0.0 without your fix -> bug
- 3.1.1 without your fix -> no bug
- 3.0.0 and above with your fix -> no bug 

I don't think it comes from the `_locator` function itself but from bbox adjustments made by matplotlib which seem to differ between versions. I spent some time trying to dig in the matplotlib code, but I'm still not sure why there is a difference here between 3.0.0 and 3.1.0...

But in any case, the fix you proposed makes more sense to me than the hardcoded `0.3` multiplier. In addition, it works as intended with matplotlib 3.0.0 and 3.1.0, so I'd vote for this change.
